### DISCUSSION TITLE
[Gluon with estimator] loss is unchanged?

### DISCUSSION THREAD

**Discussion by saicoco** [apache/incubator-mxnet#16298] [Q&A]
Note: Providing complete information in the most concise form is the best way to get help. This issue template serves as the checklist for essential information to most of the technical issues and bug reports. For non-technical issues and feature requests, feel free to present the information in what you believe is the best form.

For Q & A and discussion, please start a discussion thread at https://discuss.mxnet.io 

## Description
(Brief description of the problem in no more than 2 sentences.)

## Environment info (Required)
kaggle
```
What to do:
train with steel defect dataset by unet based resnet18_v2

```

Package used (Python/R/Scala/Julia):
I'm using python

For Scala user, please provide:
1. Java version: (`java -version`)
2. Maven version: (`mvn -version`)
3. Scala runtime if applicable: (`scala -version`)

For R user, please provide R `sessionInfo()`:

## Build info (Required if built from source)

Compiler (gcc/clang/mingw/visual studio):

MXNet commit hash:
(Paste the output of `git rev-parse HEAD` here.)

Build config:
(Paste the content of config.mk, or the build command.)

## Error Message:
(Paste the complete error message, including stack trace.)
```
[Epoch 0][Batch 0][Samples 16] time/batch: 0.475s train dice_coefficent: 0.0003, train ce_loss: 0.6932
[Epoch 0][Batch 0][Samples 16] time/batch: 0.475s train dice_coefficent: 0.0003, train ce_loss: 0.6932
[Epoch 0][Batch 0][Samples 16] time/batch: 0.475s train dice_coefficent: 0.0003, train ce_loss: 0.6932
[Epoch 0][Batch 1][Samples 32] time/batch: 0.405s train dice_coefficent: 0.0013, train ce_loss: 0.6932
[Epoch 0][Batch 1][Samples 32] time/batch: 0.405s train dice_coefficent: 0.0013, train ce_loss: 0.6932
[Epoch 0][Batch 1][Samples 32] time/batch: 0.405s train dice_coefficent: 0.0013, train ce_loss: 0.6932
[Epoch 0][Batch 2][Samples 48] time/batch: 0.349s train dice_coefficent: 0.0021, train ce_loss: 0.6932
[Epoch 0][Batch 2][Samples 48] time/batch: 0.349s train dice_coefficent: 0.0021, train ce_loss: 0.6932
[Epoch 0][Batch 2][Samples 48] time/batch: 0.349s train dice_coefficent: 0.0021, train ce_loss: 0.6932
[Epoch 0][Batch 3][Samples 64] time/batch: 0.324s train dice_coefficent: 0.0016, train ce_loss: 0.6932
[Epoch 0][Batch 3][Samples 64] time/batch: 0.324s train dice_coefficent: 0.0016, train ce_loss: 0.6932
[Epoch 0][Batch 3][Samples 64] time/batch: 0.324s train dice_coefficent: 0.0016, train ce_loss: 0.6932
[Epoch 0][Batch 4][Samples 80] time/batch: 0.302s train dice_coefficent: 0.0016, train ce_loss: 0.6932
[Epoch 0][Batch 4][Samples 80] time/batch: 0.302s train dice_coefficent: 0.0016, train ce_loss: 0.6932
[Epoch 0][Batch 4][Samples 80] time/batch: 0.302s train dice_coefficent: 0.0016, train ce_loss: 0.6932
[Epoch 0][Batch 5][Samples 96] time/batch: 0.277s train dice_coefficent: 0.0016, train ce_loss: 0.6932
[Epoch 0][Batch 5][Samples 96] time/batch: 0.277s train dice_coefficent: 0.0016, train ce_loss: 0.6932
[Epoch 0][Batch 5][Samples 96] time/batch: 0.277s train dice_coefficent: 0.0016, train ce_loss: 0.6932
[Epoch 0][Batch 6][Samples 112] time/batch: 0.292s train dice_coefficent: 0.0014, train ce_loss: 0.6932
[Epoch 0][Batch 6][Samples 112] time/batch: 0.292s train dice_coefficent: 0.0014, train ce_loss: 0.6932
[Epoch 0][Batch 6][Samples 112] time/batch: 0.292s train dice_coefficent: 0.0014, train ce_loss: 0.6932
[Epoch 0][Batch 7][Samples 128] time/batch: 0.281s train dice_coefficent: 0.0014, train ce_loss: 0.6932
[Epoch 0][Batch 7][Samples 128] time/batch: 0.281s train dice_coefficent: 0.0014, train ce_loss: 0.6932
[Epoch 0][Batch 7][Samples 128] time/batch: 0.281s train dice_coefficent: 0.0014, train ce_loss: 0.6932
[Epoch 0][Batch 8][Samples 144] time/batch: 0.274s train dice_coefficent: 0.0015, train ce_loss: 0.6932
[Epoch 0][Batch 8][Samples 144] time/batch: 0.274s train dice_coefficent: 0.0015, train ce_loss: 0.6932
[Epoch 0][Batch 8][Samples 144] time/batch: 0.274s train dice_coefficent: 0.0015, train ce_loss: 0.6932
[Epoch 0][Batch 9][Samples 160] time/batch: 0.299s train dice_coefficent: 0.0017, train ce_loss: 0.6932
[Epoch 0][Batch 9][Samples 160] time/batch: 0.299s train dice_coefficent: 0.0017, train ce_loss: 0.6932
[Epoch 0][Batch 9][Samples 160] time/batch: 0.299s train dice_coefficent: 0.0017, train ce_loss: 0.6932
[Epoch 0][Batch 10][Samples 176] time/batch: 0.275s train dice_coefficent: 0.0016, train ce_loss: 0.6932
[Epoch 0][Batch 10][Samples 176] time/batch: 0.275s train dice_coefficent: 0.0016, train ce_loss: 0.6932
[Epoch 0][Batch 10][Samples 176] time/batch: 0.275s train dice_coefficent: 0.0016, train ce_loss: 0.6932
[Epoch 0][Batch 11][Samples 192] time/batch: 0.286s train dice_coefficent: 0.0015, train ce_loss: 0.6932
[Epoch 0][Batch 11][Samples 192] time/batch: 0.286s train dice_coefficent: 0.0015, train ce_loss: 0.6932
[Epoch 0][Batch 11][Samples 192] time/batch: 0.286s train dice_coefficent: 0.0015, train ce_loss: 0.6932
[Epoch 0][Batch 12][Samples 208] time/batch: 0.300s train dice_coefficent: 0.0014, train ce_loss: 0.6932
```

## Minimum reproducible example
(If you are using your own code, please provide a short script that reproduces the error. Otherwise, please provide link to the existing example.)

## Steps to reproduce
(Paste the commands you ran that produced the error.)

1. run for this  https://www.kaggle.com/jiageng/mxnet-gluon-baseline
2.

## What have you tried to solve it?

1. change `mx.init.Normal(sigma=0.02)` to `mx.init.Xariver()`
2. change learning rate

[a comment by an automated account is not shown]

**Comment by lanking520** [comment]
@saicoco Could you please share some piece of code you run?

Note: If you change the learning rate too small, it will make the training hard to converge

**Comment by saicoco** [comment]
@lanking520 Here is my network:
```python
import mxnet as mx
from mxnet.gluon import data, HybridBlock, nn
import pandas as pd
import cv2
import os
import numpy as np
from mxnet.gluon.data.vision import transforms
from mxnet.gluon.model_zoo import vision
from mxnet.lr_scheduler import CosineScheduler
from mxnet.gluon import loss, Trainer
from mxnet import autograd
import random
from PIL import Image, ImageOps, ImageFilter
from mxnet import nd as F, lr_scheduler as lrs
from mxnet.gluon.contrib.estimator import Estimator

class SteelDataset(data.Dataset):
    def __init__(self, df, img_dir, debug=False):
        
        self.train_df = df
        self.root_dir = img_dir
        self.transform = transforms.Compose(
            [
                transforms.ToTensor(),
                transforms.Normalize(
                    mean=(0.485, 0.456, 0.406),
                   std=(0.229, 0.224, 0.225)
                )
            ]
        )
        
        self.debug = debug
        
    def __getitem__(self, i):
        if self.debug:
            curr_df = self.train_df.head(20)
        masks = np.zeros((4, 256, 1600), np.uint8)
        img_names = []
        item = self.train_df.iloc[i, :]
        img_name = item['ImageId']
        for j in range(4):
            curr_item = item["e{}".format(j+1)]
            if len(curr_item) > 0:
                rle_pixels = curr_item
                label = rle_pixels.split(" ")
                positions = list(map(int, label[0::2]))
                length = list(map(int, label[1::2]))
                mask = np.zeros(256 * 1600, dtype=np.uint8)
                for pos, le in zip(positions, length):
                    mask[pos - 1:(pos + le - 1)] = 1
                masks[j, :, :] = mask.reshape(256, 1600, order='F')
                
        if self.debug:
            oimg = cv2.imread(os.path.join(self.root_dir, img_name))[:, :, ::-1]
            img = F.array(oimg)
        else:
            img = mx.image.imread(os.path.join(self.root_dir, img_name))
        img = self.transform(img)
        if self.debug:
            return img, F.array(masks[:, ::4, ::4]), oimg, masks, curr_df
        else:
            return img, F.array(masks[:, ::4, ::4])
        
    def __len__(self):
        return len(self.train_df)

class convbn_block(HybridBlock):
    def __init__(self, in_channels, prefix):
        super().__init__(prefix=prefix)
        convbn = nn.HybridSequential(prefix=prefix)
        with convbn.name_scope():
            convbn.add(nn.BatchNorm())
            convbn.add(nn.Activation('relu'))
            convbn.add(nn.Conv2D(in_channels//4, 1))
            convbn.add(nn.BatchNorm())
            convbn.add(nn.Activation('relu'))
            convbn.add(nn.Conv2D(in_channels//4, 3, padding=(1, 1)))
            convbn.add(nn.BatchNorm())
            convbn.add(nn.Activation('relu'))
            convbn.add(nn.Conv2D(in_channels, 1))
        self.bk = convbn
    def hybrid_forward(self, F, x):
        return self.bk(x)


class SteelUnet(HybridBlock):
    
    def __init__(self, n_classes=4, ctx=mx.cpu()):
        super().__init__()
        # mb2 = vision.mobilenet_v2_1_0(pretrained=True, ctx=ctx)
        # # 128x128, 64x64, 32x32, 16x16 if input_size is 512x512
        # self.features = [mb2.features[:5],  mb2.features[5:8], mb2.features[8:15], mb2.features[15:19]]

        res18 = vision.resnet18_v2(pretrained=True, ctx=ctx)
        features = res18.features
#         for param in features.collect_params().values(): 
#             param.grad_req='null'
        features = [features[:5], features[5:7], features[7:8], features[8:9]]

        self.features = nn.HybridSequential(prefix='features')
        for f in features:
            self.features.add(f)
            
        self.convbn1 = convbn_block(512, 'convbn1')
        self.convbn2 = convbn_block(256, 'convbn2')
        self.convbn3 = convbn_block(128, 'convbn3')
        self.convbn4 = convbn_block(64, 'convbn4')
        
        self.head = nn.Conv2D(n_classes, 1, prefix='convbn_head')
        
#         self.convbn1.initialize(ctx=ctx)
#         self.convbn1.collect_params().setattr('lr_mult', 10)
#         self.convbn2.initialize(ctx=ctx)
#         self.convbn2.collect_params().setattr('lr_mult', 10)
#         self.convbn3.initialize(ctx=ctx)
#         self.convbn3.collect_params().setattr('lr_mult', 10)
#         self.convbn4.initialize(ctx=ctx)
#         self.convbn4.collect_params().setattr('lr_mult', 10)
#         self.head.initialize(ctx=ctx)
#         self.head.collect_params().setattr('lr_mult', 10)
        
    def hybrid_forward(self, F, x):
        
        c1 = self.features[0](x)
        c2 = self.features[1](c1)
        c3 = self.features[2](c2)
        c4 = self.features[3](c3)
        c4 = self.convbn1(c4)

        upsample_c4 = F.contrib.BilinearResize2D(c4, scale_height=2, scale_width=2)
        c3 = F.Activation(c3, act_type='relu')
        c3 = self.convbn2(F.concat(upsample_c4, c3, dim=1))
        upsample_c3 = F.contrib.BilinearResize2D(c3, scale_height=2, scale_width=2)
        c2 = F.Activation(c2, act_type='relu')
        c2 = self.convbn3(F.concat(upsample_c3, c2, dim=1))
        upsample_c2 = F.contrib.BilinearResize2D(c2, scale_height=2, scale_width=2)
        c1 = F.Activation(c1, act_type='relu')
        c1 = self.convbn4(F.concat(upsample_c2, c1, dim=1))

        out = F.sigmoid(self.head(c1))
        print(out.max(), out.min())
        return out

```

I add relu for bolcks of resnet18_v2, however it doesn't work..The task is about semantic segmentation, I use BCEloss for it, dataloader is correct, I think there is problem in network
